tests/Mvp.Selections.Api.Tests/Services/LicenseServiceTests.cs

```diff
@@ -0,0 +1,223 @@
+            };
+
+            // Act
+            var result = await _sut.AssignLicenseToUserAsync(request, licenseId);
+
+            // Assert
+            result.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+            result.Messages.Should().Contain("Email is required");
+        }
+
+        [Fact]
+        public async Task AssignUserAsync_WithValidUserAndLicense_ReturnsSuccess()
+        {
+            // Arrange
+            var licenseId = Guid.NewGuid();
+            var email = "test@example.com";
+            var license = new Domain.License(licenseId)
+            {
+                LicenseContent = "test content",
+                ExpirationDate = DateTime.UtcNow.AddDays(30)
+            };
+
+            var user = new User(Guid.NewGuid())
+            {
+                Email = email
+            };
+
+            var request = new AssignUserToLicense
+            {
+                Email = email
+            };
+
+            _licenseRepository.GetLicenseAsync(licenseId)
+                .Returns(license);
+
+            _userRepository.GetAllAsync(email: email)
+                    .Returns(new List<User> { user });
+
+            _licenseRepository.UserHasTitleForYearAsync(user, DateTime.Now.Year)
+                .Returns(true);
+
+            _licenseRepository.AssignLicenseToUserAsync(Arg.Any<Domain.License>())
+                    .Returns(Task.CompletedTask);
+
+            _licenseRepository.SaveChangesAsync()
+                .Returns(Task.CompletedTask);
+
+            // Act
+            var result = await _sut.AssignLicenseToUserAsync(request, licenseId);
+
+            // Assert
+            result.StatusCode.Should().Be(HttpStatusCode.OK);
+            result.Result.Should().Be(license);
+
+            await _licenseRepository.Received(1)
+                .AssignLicenseToUserAsync(Arg.Is<Domain.License>(l => l.AssignedUserId == user.Id));
+        }
+
+        [Fact]
+        public async Task AssignUserAsync_WithInvalidLicenseId()
+        {
+            // Arrange
+            var nonExistentLicenseId = Guid.NewGuid();
+            var email = "test@example.com";
+
+            Guid licenseId = Guid.NewGuid();
+            var request = new AssignUserToLicense
+            {
+                Email = email
+            };
+
+            _licenseRepository.GetLicenseAsync(nonExistentLicenseId)
+                .Returns((Domain.License?)null);
+
+            // Act
+            var result = await _sut.AssignLicenseToUserAsync(request, licenseId);
+
+            // Assert
+            result.StatusCode.Should().Be(HttpStatusCode.BadRequest);
+            result.Messages.Should().Contain("License not found");
+            result.Result.Should().BeNull();
+
+            await _licenseRepository.DidNotReceive()
+                .AssignLicenseToUserAsync(Arg.Any<Domain.License>());
+        }
+
+        [Fact]
+        public async Task ZipUploadAsync_WithValidFile_ReturnsSuccessResult()
```

Comment:
We have the happy path test here, but are missing tests for the two possible failure exits on the `ZipUploadAsync` method